File: Ruby/06_nonjson_ops.rb
#!/usr/bin/env ruby

require "rubygems"
require "couchbase"
require "rainbow"
require "pp"

system 'clear'
puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts "## Couchbase Non-JSON Ops".bright.foreground(:red)
puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts

cb = Couchbase.connect

puts "Store Strings, Floats, Integers".bright.foreground(:red)
puts

cb.set("store_strings", "The quick brown fox jumped over the lazy dog.", :ttl => 10)
cb.set("store_floats", 3.14159265358979, :ttl => 10)
cb.set("store_integers", -42, :ttl => 10)

puts "cb.get(\"store_strings\")" + " => ".bright + "\"" + cb.get("store_strings") + "\""
puts cb.get("store_strings").class.to_s
puts

puts "cb.get(\"store_floats\")" + " => ".bright + cb.get("store_floats").to_s
puts cb.get("store_floats").class.to_s
puts

puts "cb.get(\"store_integers\")" + " => ".bright + cb.get("store_integers").to_s
puts cb.get("store_integers").class.to_s
puts

puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts "Append/Prepend Ops".bright.foreground(:red)
puts

puts "Create a list, start with one item".bright
puts

cb.set("mylist", "oranges", :format => :plain, :ttl => 30)
puts "cb.get(\"mylist\")" + " => ".bright + cb.get("mylist").to_s
puts

puts "Add an item...\n\n"
cb.prepend("mylist", "apples,", :format => :plain)
puts "cb.get(\"mylist\")" + " => ".bright + cb.get("mylist").to_s
puts

puts "Add an item...\n\n"
cb.append("mylist", ",bananas", :format => :plain)
puts "cb.get(\"mylist\")" + " => ".bright + cb.get("mylist").to_s
puts

puts "Add an item...\n\n"
cb.append("mylist", ",lemons", :format => :plain)
puts "cb.get(\"mylist\")" + " => ".bright + cb.get("mylist").to_s
puts

puts "Add an item...\n\n"
cb.append("mylist", ",grapes", :format => :plain)
puts "cb.get(\"mylist\")" + " => ".bright + cb.get("mylist").to_s
puts


puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts "Store Binary Blobs".bright.foreground(:red)
puts

# delete output if it exists
if File.file?("./output.jpg")
  File.delete("./output.jpg")
end

source = "../../Binary/lolcat.jpg"
binary_img = nil

puts "Read jpg from disk".bright
puts
# read from disk
File.open(source, "r") {|f|
  binary_img = f.read
}
# alternate way to read binary files
# binary_img = File.open(source,"rb") {|io| io.read}

puts "Store file in Couchbase, :format => :plain is important to set for binary".bright
puts
cb.set("img", binary_img, :ttl => 10, :format => :plain)


puts "Retrieve from Couchbase".bright
puts
img_cb = cb.get("img")


puts "Save image from Couchbase to disk (checkout output.jpg)".bright
f = File.new("./output.jpg", "w")
f.write(img_cb)
f.close

puts
puts "--------------------------------------------------------------------------".bright.foreground(:red)
puts

File: PHP/06_nonjson_ops.php
<?php

echo "--------------------------------------------------------------------------\n";
echo "\tCouchbase Non-JSON Ops\n";
echo "--------------------------------------------------------------------------\n";

if (file_exists("./output.jpg") ) {
	unlink("./output.jpg");	
}


$cb = new Couchbase();


echo "Store Strings, Floats, Integers\n";
$cb->set("store_strings", "The quick brown fox jumped over the lazy dog.", 10);
$cb->set("store_floats", 3.14159265358979, 10);
$cb->set("store_integers", -42, 10);

echo "get 'store_strings => '";
echo $cb->get("store_strings");
echo "\n\n";

echo "get 'store_floats' => ";
echo $cb->get("store_floats");
echo "\n\n";

echo "get 'store_integers' => ";
echo $cb->get("store_integers");
echo "\n\n";


echo  "--------------------------------------------------------------------------\n";
echo  "Append/Prepend Ops\n";

echo "Create a list, start with one item\n";
$cb->set("mylist", "oranges", 30);

echo "Add an item...\n\n";
$cb->prepend("mylist", "apples,");

echo "Add an item...\n\n";
$cb->append("mylist", ",bananas");

echo "Add an item...\n\n";
$cb->append("mylist", ",lemons");

echo "Add an item...\n\n";
$cb->append("mylist", ",grapes");

echo "get 'mylist' => ";
echo $cb->get("mylist");
echo "\n\n";


echo "--------------------------------------------------------------------------\n";
echo "Store Binary Blobs\n";

// read file
echo "\nRead file\n";
$image = file_get_contents('../Binary/lolcat.jpg');
$image_data = base64_encode($image);
$cb->set("img", $image_data, 10);

$to_save = base64_decode(  $cb->get("img") );

file_put_contents("./output.jpg", $to_save);



echo "\n--------------------------------------------------------------------------\n";


?>